cmd/kubernetes-discovery/discoverysummarizer/types.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package discoverysummarizer
+
+// List of servers from which group versions should be summarized.
+type FederatedServerList struct {
+	Servers []FederatedServer `json:"servers"`
+}
+
+// Information about each individual server, whose group versions needs to be summarized.
+type FederatedServer struct {
+	// The address that summarizer can reach to get discovery information from the server.
+	// This can be hostname, hostname:port, IP or IP:port.
+	ServerAddress string `json:"serverAddress"`
+	// The list of paths where server exposes group version discovery information.
+	// Summarizer will use these paths to figure out group versions supported by this server.
+	GroupVersionDiscoveryPaths []GroupVersionDiscoveryPath `json:"groupVersionDiscoveryPaths"`
+}
```

Comment:
I think that's OK for the initial commit but it needs to eventually run in a pod in the cluster, so it will need to auth to each federated server. I guess you can use the inClusterConfig? No guarantee that will work for all of the servers though, since we don't require a unified auth domain (despite it being a really good idea in practice).